amplipi/app.py

```diff
@@ -681,6 +685,35 @@ def create_app(mock_ctrl=None, mock_streams=None, config_file=None, delay_saves=
   get_ctrl().reinit(settings, change_notifier=notify_on_change)
   return app
 
+def advertise_service(port):
+  """ Advertise the AmpliPi api via zerconf, can be verified with 'avahi-browse -ar' """
+  hostname = f'{gethostname()}.local'
+  url = f'http://{hostname}'
+  if port != 80:
+    url += f':{port}'
+  info = ServiceInfo(
+    "_http._tcp.local.",
+    "amplipi-api._http._tcp.local.", # this is named AmpliPi-api to distinguish from the common Spotify/Airport name of AmpliPi
+    addresses=[inet_aton("127.0.0.1")],
+    port=port,
+    properties={
+      # standard info
+      'path': '/api/',
+      # extra info - for interfacing
+      'name': 'AmpliPi',
+      "vendor": 'MicroNova',
+      'version': utils.detect_version(),
+      # extra info - for user
+      'web_app': url,
+      'documentation': f'{url}/doc'
+    },
+    server=f'{hostname}.', # Every other service has a trailing '.' here. If it is missing Zeroconf's example/browser.py fails to process this.
```

Comment:
See the definition of an SRV record for why a trailing dot is required for the service and target https://en.wikipedia.org/wiki/SRV_record